### PR TITLE
Feat/detach zone

### PR DESCRIPTION
**Pull request by marie-j**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-3729
| License          | BSD 3-Clause

## Description

- Reapply zone detachment : This reverts commit 21f6de0.
- Prevent dashboard crashing

### CODE REVIEW

File: packages/manager/apps/web/client/app/domain/zone/detach/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "web_domain_zone_detach": "Délier ma zone DNS",
+  "web_domain_zone_detach_description": "Délier cette zone DNS vous permet de la renouveler de manière indépendante de votre domaine. Ceci vous permet de continuer d'utiliser votre zone DNS pour un nom de domaine déposé dans un autre registre.",
+  "web_domain_zone_detach_error": "Une erreur est survenue lors du détachement de la zone. {{ message }}",
+  "web_domain_zone_detach_success_autopay": "Le détachement de la zone a bien été réalisée"
```
Comment (FredericEspiau) [inline packages/manager/apps/web/client/app/domain/zone/detach/translations/Messages_fr_FR.json:5]:
```suggestion
  "web_domain_zone_detach_success_autopay": "Le détachement de la zone a bien été réalisé"
```

File: packages/manager/apps/web/client/app/domain/zone/detach/detach.module.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import 'angular-translate';
+import productOffers from '@ovh-ux/manager-product-offers';
+import component from './detach.component';
+import routing from './detach.routing';
+
+const moduleName = 'ovhManagerWebDomainZoneDetach';
+
+angular
+  .module(moduleName, ['pascalprecht.translate', productOffers])
```
Comment (FredericEspiau) [inline packages/manager/apps/web/client/app/domain/zone/detach/detach.module.js:9]: Routing, translation, uikit

**Comment by marie-j** [comment]
@FredericEspiau Thanks for reviewing and pointing out issues. The issues are related to reapplying code that was reverted due to an API issue. I'll update the translations but I'm not making further changes to code that was reviewed and validated by QC

**Comment by FredericEspiau** [comment]
@marie-j it's only adding missing dependencies 😢

**Review by JeremyDec — DISMISSED** [review]

**Comment by antleblanc** [comment]
@marie-j It seems that this ticket has been unblocked.
Could you please resolve existing conflicts in order to re-run the verification process?

Thank you!

**Comment by ovh-cds** [comment]
CDS Report test#16517.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✔

**Comment by marie-j** [comment]
Rebase on develop to fix conflicts

File: packages/manager/apps/web/client/app/domain/zone/detach/translations/Messages_fr_FR.json (hunk 1 above)
Comment (antleblanc) [reply packages/manager/apps/web/client/app/domain/zone/detach/translations/Messages_fr_FR.json:5]: @marie-j Do you plan to integrate this typo?

File: packages/manager/apps/web/client/app/domain/zone/zone.routing.js [hunk 3]
```diff
@@ -1,6 +1,31 @@
...
+    $q,
+    ovhManagerProductOffersDetachService,
+    serviceInfos,
+  ) =>
+    (serviceInfos
+      ? ovhManagerProductOffersDetachService.getAvailableDetachPlancodes(
+          serviceInfos.serviceId,
+        )
+      : $q.resolve([])
+    ).catch(() => $q.resolve([])),
```
Comment (JeremyDec) [inline packages/manager/apps/web/client/app/domain/zone/zone.routing.js:16]: **nitpick**:
```suggestion
      : []
    ).catch(() => []),
```

**Review by JeremyDec — APPROVED** [review]

**Comment by marie-j** [comment]
Rebased on develop

**Comment by antleblanc** [comment]
> Rebased on develop

Could you please re-do? (new conflicts appears since then)
Thank you!

**Pull request closed without merging**